packages/async-flow/src/types.d.ts

```diff
@@ -0,0 +1,252 @@
+ *
+ * Specifically, Promise return values are converted to Vows.
+ */
+export type HostOf<F> = F extends (...args: infer A) => Promise<infer R>
+  ? (...args: A) => Vow<R extends Passable ? R : HostInterface<R>>
+  : F;
+
+export type PreparationOptions = {
+  vowTools?:
+    | {
+        when: <
+          T,
+          TResult1 = import('@agoric/vow').EUnwrap<T>,
+          TResult2 = never,
+        >(
+          specimenP: T,
+          onFulfilled?:
+            | ((
+                value: import('@agoric/vow').EUnwrap<T>,
+              ) => TResult1 | PromiseLike<TResult1>)
+            | undefined,
+          onRejected?:
+            | ((reason: any) => TResult2 | PromiseLike<TResult2>)
+            | undefined,
+        ) => Promise<TResult1 | TResult2>;
+        watch: <
+          T = any,
+          TResult1 = T,
+          TResult2 = never,
+          C extends any[] = any[],
+        >(
+          specimenP: import('@agoric/vow').EVow<T>,
+          watcher?:
+            | import('@agoric/vow').Watcher<T, TResult1, TResult2, C>
+            | undefined,
+          ...watcherArgs: C
+        ) => Vow<
+          Exclude<TResult1, void> | Exclude<TResult2, void> extends never
+            ? TResult1
+            : Exclude<TResult1, void> | Exclude<TResult2, void>
+        >;
+        makeVowKit: <T>() => import('@agoric/vow').VowKit<T>;
+        allVows: (
+          maybeVows: import('@agoric/vow').EVow<unknown>[],
+        ) => Vow<any[]>;
+        asVow: <T extends unknown>(
+          fn: (
+            ...args: any[]
+          ) =>
+            | Vow<Awaited<T>>
+            | Awaited<T>
+            | import('@agoric/vow').PromiseVow<T>,
+        ) => Vow<Awaited<T>>;
+        asPromise: import('@agoric/vow').AsPromiseFunction;
+        retriable: <F extends (...args: any[]) => Promise<any>>(
+          fnZone: import('@agoric/base-zone').Zone,
+          name: string,
+          fn: F,
+        ) => F extends (...args: infer Args) => Promise<infer R>
+          ? (...args: Args) => Vow<R>
+          : never;
+      }
+    | undefined;
+  makeLogStore?: (() => LogStore) | undefined;
+  makeBijection?: (() => Bijection) | undefined;
+  endowmentTools?:
+    | {
+        prepareEndowment: (
+          zone: import('@agoric/base-zone').Zone,
+          tag: string,
+          e: unknown,
+        ) => any;
+        unwrap: (wrapped: any, guestWrapped: any) => any;
+      }
+    | undefined;
+};
+export type OutcomeKind = 'return' | 'throw';
+
+export type Outcome =
+  | {
+      kind: 'return';
+      result: any;
+    }
+  | {
+      kind: 'throw';
+      problem: any;
+    };
+
+export type Ephemera<S extends WeakKey = WeakKey, V extends unknown = any> = {
+  for: (self: S) => V;
+  resetFor: (self: S) => void;
+};
+
+/**
+ * This is the typedef for the membrane log entries we currently implement.
+ * See comment below for the commented-out typedef for the full
```

Comment:
As translated to ts, is the term "typedef" now stale?

Reply:
it's the same entity, regardless of syntax. I suppose technically in both cases it's a "type alias". I'll revise.